src/model/src/ModelTransformers.cpp

```diff
@@ -850,6 +868,451 @@ bool createReducedModelAndChangeLinkFrames(const Model& fullModel,
     return ok;
 }
 
+// New vector-based implementation for joints with multiple position coordinates
+bool createReducedModelAndChangeLinkFrames(const Model& fullModel,
+                                           const std::vector< std::string >& jointsInReducedModel,
+                                           Model& reducedModel,
+                                           const std::unordered_map<std::string, std::vector<double>>& removedJointPositions,
+                                           const std::unordered_map<std::string, iDynTree::Transform>& newLink_H_oldLink,
+                                           bool addOriginalLinkFrameWith_original_frame_suffix,
+                                           bool includeAllAdditionalFrames,
+                                           const std::vector<std::string>& allowedAdditionalFrames)
+{
+    // We use the default traversal for deciding the base links of the reduced model
+    Traversal fullModelTraversal;
+    fullModel.computeFullTreeTraversal(fullModelTraversal);
+
+    // To compute the inertia and all the other properties of the links of
+    // the reduced model, we need to build a SubModelDecomposition of the full
+    // model, in which every SubModel corresponds to a link in the reduced model
+    SubModelDecomposition subModels;
+
```

Comment:
Can we avoid the duplication? This function completly duplicates the existing `createReducedModelAndChangeLinkFrames` function.